modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2Utils.java

```diff
@@ -437,6 +458,45 @@ public static void setupConnection(
         s.setForceJoinOrder(enforceJoinOrder);
         s.setJoinBatchEnabled(distributedJoins);
         s.setLazyQueryExecution(lazy);
+
+        QueryContext oldCtx = (QueryContext)s.getVariable(QCTX_VARIABLE_NAME).getObject();
+
+        assert oldCtx == null || oldCtx == qctx : oldCtx;
+
+        s.setVariable("IGNITE_QUERY_CONTEXT", new ValueRuntimeSimpleObject<>(qctx));
```

Comment:
Do you mean QCTX_VARIABLE_NAME here instead of "IGNITE_QUERY_CONTEXT" ?
BTW 'QCTX_VARIABLE_NAME' constant is not equals to "IGNITE_QUERY_CONTEXT"
Is it a typo, I've missed smth?